user.go

```diff
@@ -37,53 +110,87 @@ func (h UserHandler) AddUserHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	
+	user.ID = primitive.NewObjectID()
 
-	response, err := json.Marshal(user)
+	insertResult, err := h.Collection.InsertOne(context.Background(), user)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	
+	insertResponse, err := json.Marshal(insertResult.InsertedID)
 
 	w.Header().Set("Content-Type", "application/json")
-	w.Write(response)
+	w.Write(insertResponse)
 }
 
 // UpdateUserHandler Gets a PUT request with user's data in the request's body
 // and updates it in the database and responds with the updated user's data
 func (h UserHandler) UpdateUserHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	userID := vars["id"]
-	if userID == "" {
-
+	userOid, err := primitive.ObjectIDFromHex(userID)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
-
+	
 	decoder := json.NewDecoder(r.Body)
-
+	
 	var user User
-	err := decoder.Decode(&user)
+	err = decoder.Decode(&user)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
+	filter := bson.D{bson.E{ Key: "_id", Value: userOid}}
+	update := bson.D{
+		{ Key: "$set", Value: bson.D{{ Key: "name", Value: user.Name}}},
+		{ Key: "$set", Value: bson.D{{ Key: "admin", Value: user.Admin}}},
```

Comment:
In a case of a partial user data, what will happen here? isn't it optional that user data will be deleted?

Reply:
Yes...I should use ,omitempty